xapian-core/cluster/cluster.cc

```diff
@@ -246,12 +236,15 @@ Point::initialize(TermListGroup &tlg, const Document &doc_) {
     for (TermIterator it = doc.termlist_begin(); it != doc.termlist_end(); ++it) {
 	doccount wdf = it.get_wdf();
 	string term = *it;
-	if (wdf < 1 || term[0]=='Z')
+	double termfreq = tlg.get_termfreq(term);
+	/** If term is not a stemmed term or the term indexes only one
+	 *  one document, do not compute TF-IDF scores for them
+	 */
+	if (wdf < 1 || term[0] != 'Z' || termfreq <= 1)
```

Comment:
The TLG only calculates the term frequencies for terms by checking their occurrences in various documents. It doesn't deal with creating the document vector. It is only when we initialize the Point, that we loop through the termlist of the document and create the document vector. So it would be rather hard to filter out terms in the TLG itself according to me.
Couldn't we extend the idea of filtering terms within the Point class too?